spec/Functions.tex

```diff
@@ -1041,6 +1041,37 @@ \subsection{Implicit Return Types}
 inferred return type.
 If the above requirements are not satisfied, it is an error.
 
+\section{Where Expressions}
+\label{Where_Expressions}
+\index{where@\chpl{where}}
+\index{expressions!where@\chpl{where}}
+\index{functions!where@\chpl{where}}
+
+The list of function candidates can be constrained by {\em where clauses}.  A
+where clause is specified in the definition of a
+function~(\rsec{Function_Definitions}).  When a function is resolved, the
+expression in the where clause must be a parameter expression and must
+evaluate to either \chpl{true} or \chpl{false}.  If it evaluates to
+\chpl{false}, the function is rejected and thus is not a possible candidate
+for function resolution.
+\begin{chapelexample}{whereClause.chpl}
+Given two overloaded function definitions
+\begin{chapel}
+proc foo(x) where x.type == int { writeln("int"); }
+proc foo(x) where x.type == real { writeln("real"); }
+\end{chapel}
```

Comment:
I haven't looked at the context enough to have a strong opinion on this:  is it weird to use a generic function example in this section (i.e., before the "Generics" chapter?)  I don't find the presence of it here inherently off-putting, but I have a pretty low bar when it comes to such things.  If it does seem weird, I might use a concrete function overload here -- say one based on whether or not a config param is true v. false and move this example back to the Generics chapter with a perfunctory section that essentially says something like "Where clauses are particularly useful in the context of generic functions since they can be used to reason about the argument types."  But again, I don't feel insistent on this if the current approach seems OK.